rust_snuba/rust_arroyo/src/backends/kafka/mod.rs

```diff
@@ -78,8 +78,14 @@ impl KafkaConsumerState {
     }
 }
 
-fn create_kafka_message(msg: BorrowedMessage) -> BrokerMessage<KafkaPayload> {
-    let topic = Topic::new(msg.topic());
+fn create_kafka_message(topics: &[Topic], msg: BorrowedMessage) -> BrokerMessage<KafkaPayload> {
+    let topic = msg.topic();
+    let topic = *topics
+        .iter()
+        .find(|t| t.as_str() == topic)
+        .unwrap_or_else(|| {
+            panic!("Received message for topic `{topic}` that we never subscribed to");
+        });
```

Comment:
`unwrap_or_else` with a `panic` inside is more naturally written with `let else`, I think.